matplotcheck/raster.py

```diff
@@ -140,27 +184,12 @@ def assert_legend_accuracy_classified_image(
         # Check that each legend entry label is in one of all_label_options
         assert len([val for val in legend_dict.values() if val]) == len(
             all_label_options
-        ), "Incorrect legend labels"
+        ), (
+            "Number of label options provided doesn't match the number of"
+            " labels found in the image."
+        )
 
-        # Create two copies of image array, one filled with the plot data class
-        # labels (im_data_labels) and the other with the expected labels
-        # (im_expected_labels)
-        im_class_dict = {}
-        for val in np.unique(im_data):
-            im_class_dict[val] = legend_dict[im_cmap(im.norm(val))]
-        im_data_labels = [
-            [im_class_dict[val] for val in row] for row in im_data.data
-        ]
-        im_expected_labels = [
-            [all_label_options[val][0] for val in row] for row in im_expected
-        ]
-
-        # Check that expected and actual labels match up
-        assert np.array_equal(
-            im_data_labels, im_expected_labels
-        ), "Incorrect legend to data relation"
-
-        # IMAGE TESTS/HELPER FUNCTIONS
+        return legend_dict
```

Comment:
so i'm curious about this part. i vaguely remember checking that the colormap matched the labels and wonder if that is still happening? i remember working with kristen on this but it was a long time ago.